sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/DataSourceV2AnalysisSuite.scala

```diff
@@ -248,6 +248,68 @@ abstract class DataSourceV2AnalysisBaseSuite extends AnalysisTest {
 
   def byPosition(table: NamedRelation, query: LogicalPlan): LogicalPlan
 
+  test("SPARK-33136: output resolved on complex types for V2 write commands") {
```

Comment:
Is this adding a test case for SPARK-33136 in the PR for SPARK-33360, @cloud-fan ?

Reply:
Isn't it natural to make this as a follow-up or separate PR instead of having in this PR, `Simplify DS v2 write resolution`?